xbmc/video/VideoDatabase.cpp

```diff
@@ -2236,6 +2238,10 @@ int CVideoDatabase::SetDetailsForMovie(const std::string& strFilenameAndPath, CV
       sql += PrepareSQL(", userrating = %i", details.m_iUserRating);
     else
       sql += ", userrating = NULL";
+    if (details.HasPremiered())
+      sql += PrepareSQL(", premiered = '%s'", details.GetPremiered().GetAsDBDate().c_str());
+    else
+      sql += PrepareSQL(", premiered = '%i'", details.GetYear());
```

Comment:
Why not write a string in the form of `<year>-01-01` (or whatever the DB date format looks like)? That way you don't need to do any odd handling when reading the values again.